rasa/core/agent.py

```diff
@@ -392,156 +312,110 @@ class Agent:
     def __init__(
         self,
         domain: Union[Text, Domain, None] = None,
-        policies: Union[PolicyEnsemble, List[Policy], None] = None,
-        interpreter: Optional[NaturalLanguageInterpreter] = None,
         generator: Union[EndpointConfig, NaturalLanguageGenerator, None] = None,
         tracker_store: Optional[TrackerStore] = None,
         lock_store: Optional[LockStore] = None,
         action_endpoint: Optional[EndpointConfig] = None,
         fingerprint: Optional[Text] = None,
-        model_directory: Optional[Text] = None,
         model_server: Optional[EndpointConfig] = None,
         remote_storage: Optional[Text] = None,
-        path_to_model_archive: Optional[Text] = None,
+        graph_runner: Optional[GraphRunner] = None,
+        model_path: Optional[Text] = None,
+        model_id: Optional[Text] = None,
     ):
-        # Initializing variables with the passed parameters.
-        self.domain = self._create_domain(domain)
-        self.policy_ensemble = self._create_ensemble(policies)
-
-        PolicyEnsemble.check_domain_ensemble_compatibility(
-            self.policy_ensemble, self.domain
-        )
-
-        self.interpreter = rasa.core.interpreter.create_interpreter(interpreter)
+        """Initializes an `Agent`."""
+        self.processor = None
+        self.domain = domain
+        if self.domain:
```

Comment:
This check is in my opinion outdated. "In the early days" you had to specify `utter_` in both, `actions` and `responses`. Nowadays it's enough to have them specified in `responses`. They will be added to `actions` on the fly. So what this ends up checkin is that a user didn't add a `utter_` thing to their actions while not having it in responses. And that should in my opinion be validated in the `Domain` itself instead of here.